### PR TITLE
improvement: S3C-3098 support of async/await by eslint

### PR DESCRIPTION
**Pull request by jonathan-gramain**
Add the needed parserOption.ecmaVersion=8 parameter to support
async/await syntax in other repositories.

Remove arrow-parens rule as it gives false positives when using
"async" keyword on arrow function.

### CODE REVIEW

**Comment by jonathan-gramain** [comment]
@bert-e reset

**Review by ilkescality — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -47,7 +51,7 @@ module.exports = {
         'arrow-body-style': [1, 'as-needed'],
         'prefer-rest-params': 1,
         'no-unneeded-ternary': [1, { defaultAssignment: false }],
-        'arrow-parens': [1, 'as-needed'],
+        'arrow-parens': 0,
```
Comment (rahulreddy) [inline index.js:54]: This is sneaky 🗡️

Comment (jonathan-gramain) [reply index.js:54]: It's in the commit message :slightly_smiling_face:

**Review by rahulreddy — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (rahulreddy) [reply index.js:54]: Ah, usually comments appear in the desc.

**Review by miniscruff — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (jonathan-gramain) [reply index.js:54]: Ah yes, I :snake: -ily force pushed without updating the PR description :smile:

**Comment by jonathan-gramain** [comment]
For information: I made this PR a no-op on 8.1+ branches, it does not make much sense as the version of eslint has been bumped.

**Comment by jonathan-gramain** [comment]
@bert-e approve

**Pull request merged**